Wrappers/Python/test/test_Gradient.py

```diff
@@ -567,5 +567,68 @@ def test_GradientOperator_split(self):
         numpy.testing.assert_allclose(out4[0].array, out4_gold[2].array, atol = 0.4)
         numpy.testing.assert_allclose(out4[1][0].array, out4_gold[0].array, atol = 0.4)  
         numpy.testing.assert_allclose(out4[1][1].array, out4_gold[1].array, atol = 0.4)        
-        numpy.testing.assert_allclose(out4[1][2].array, out4_gold[3].array, atol = 0.4)        
+        numpy.testing.assert_allclose(out4[1][2].array, out4_gold[3].array, atol = 0.4) 
+
+    def test_GradientOperator_for_pseudo_3D_geometries(self):   
+
+        numpy.random.seed(10)
+        ig = ImageGeometry(voxel_num_x = 10, voxel_num_y = 10, voxel_num_z=1,
+                            voxel_size_x = 0.4, voxel_size_y = 0.2, voxel_size_z=0.6)
+
+        data = ig.allocate('random')                            
```

Comment:
Instead of having a random dataset for unit testing, why don't you construct 2 phantoms with 2 levels, 0 and 1, one split horizontally and one vertically vertically? You'd know exactly that the gradient would be zero and a single line at the boundary in the other.

Reply:
Yeah, it's always better to test with known values so if things break we can easily check the values manually to debug.

Reply:
fixed